opentuner/search/technique.py

```diff
@@ -242,6 +267,104 @@ def get_enabled(args):
 
   return [t for t in techniques if t.name in args.technique]
 
+def generate_bandit_technique(args, manipulator, num_techniques=5, retry_count=3, composable_weight=2):
+  """
+  Generate a bandit by randomly selecting existing techniques or composable techniques.
+  If a composable technique is selected, the operators are then chosen
+
+  :param args: arguments to the program
+  :param manipulator: a ConfigurationManipulator used to enumerate parameters
+  :param num_techniques: max number of subtechniques in the bandit
+  :param retry_count: number of times to try getting a new technique before giving up
+  :param composable_weight: weight to increase probability of choosing a composable technique
+  """
+  from metatechniques import MetaSearchTechnique
+  from bandittechniques import AUCBanditMetaTechnique
+  techniques, composables = all_techniques(args)
+  if args.list_techniques:
+    for t in techniques:
+      print t.name
+    sys.exit(0)
+
+  # get set of parameters to consider
+  paramset = set()
+  for p in manipulator.params:
+    paramset.add(type(p))
+
+  # filter techniques to get rid of metatechniques
+  basetechniques = [t for t in techniques if not isinstance(t, MetaSearchTechnique)]
+
+  bandit_techniques = []
+  for i in range(num_techniques):
+    for j in range(retry_count):
+      # pick a technique or generate a composable
+      if random.random() < float(len(basetechniques)) / (len(basetechniques) + composable_weight*len(composables)):
+        candidate = copy.deepcopy(random.choice(basetechniques))
+      else:
+        # generate a composable technique
+        candidate = generate_composable_technique(composables, paramset)
+      if not (candidate.name in [t.name for t in bandit_techniques]):
+        bandit_techniques.append(candidate)
+        break
+
+  # make a bandit of the output list
+  return AUCBanditMetaTechnique(bandit_techniques, name="GeneratedBandit")
+
+# params = list of parameters that we have to choose operators for
+def generate_composable_technique(techniques, paramset):
+  from composableevolutionarytechniques import ComposableEvolutionaryTechnique
```

Comment:
Should we move this method to composableevolutionarytechniques so we don't have circular import messyness?